File: C++/033. Search in Rotated Sorted Array.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int search(std::vector<int>& nums, int target) {
        int i = 0, j = int(nums.size()) - 1;
        while (i <= j) {
            int m = (i + j) / 2;
            if (nums[m] == target)
                return m;
            if (nums[i] > nums[m] && (nums[i] <= target || target < nums[m]))
                j = m - 1;
            else if (nums[i] < nums[m] && nums[i] <= target && target < nums[m])
                j = m - 1;
            else
                i = m + 1;
        }
        return -1;
    }
};

File: Go/033. Search in Rotated Sorted Array.go
package main

func search(nums []int, target int) int {
	i, j := 0, len(nums) - 1
	for i <= j {
		m := (i + j ) / 2
		if nums[m] == target {
			return m
		}
		if nums[i] > nums[m] && (target >= nums[i] || target < nums[m]) {
			j = m - 1
		} else if nums[i] < nums[m] && target >= nums[i] && target < nums[m] {
			j = m - 1
		} else {
			i = m + 1
		}
	}
	return -1
}